Textfield/Textfield.jsx

```diff
@@ -64,33 +68,58 @@ class TextfieldInput extends MaterialComponent {
       }
     );
   }
+
   componentWillUnmount() {
     this.MDComponent && this.MDComponent.destroy && this.MDComponent.destroy();
   }
-  materialDom(allprops) {
+  
+  _onFocus() {
+    this.setState({
+      isFocused: true
+    });
+  }
+
+  _onBlur() {
+    this.setState({
+      isFocused: false
+    });
+  }
+
+  materialDom(allprops, { showFloatingLabel, isFocused }) {
     let { className, ...props } = allprops;
+    let labelClassName = "mdc-textfield__label";
+
+    const upgrade = props.value || isFocused;
 
-    if (props.value && this.state.showFloatingLabel) {
+    if (showFloatingLabel) {
       className = [className, "mdc-textfield--upgraded"].join(" ");
```

Comment:
sorry I might have forgot, y are we doing this by hand when `mdc-web` is already doing this for us?